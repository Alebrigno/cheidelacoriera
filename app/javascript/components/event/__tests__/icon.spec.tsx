import { configure } from "enzyme";
import * as Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });

import { EventIcon } from "..";

describe("EventIcon", () => {
  const STATUSES = [undefined, "available", "fully_booked"];
  const AUDIENCES = [undefined, "preferred", "gold"];

  STATUSES.forEach((status: string) => {
    AUDIENCES.forEach((audience: string) => {
      test(`when status: "${status}" and audience: "${audience}"`, () => {
        const statusProp = status ? { status: status } : {};
        const audienceProp = audience ? { audience: audience } : {};
        const eventIcon = new EventIcon({
          game: { ...statusProp, ...audienceProp },
        });

        expect(eventIcon.render()).toMatchSnapshot();
      });
    });
  });
});
