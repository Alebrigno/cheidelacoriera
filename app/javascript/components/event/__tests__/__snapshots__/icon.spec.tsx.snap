// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventIcon when status: "available" and audience: "gold" 1`] = `
<div
  className="pull-right"
>
  <Icon
    className="fa fa-calendar-check-o icon-green"
    message="Prenotabile"
  />
  <Icon
    className="fa fa-trophy icon-yellow"
    message="Solo Corieristi Gold"
  />
</div>
`;

exports[`EventIcon when status: "available" and audience: "preferred" 1`] = `
<div
  className="pull-right"
>
  <Icon
    className="fa fa-calendar-check-o icon-green"
    message="Prenotabile"
  />
  <Icon
    className="fa fa-star icon-yellow"
    message="Solo Corieristi Semper Fidelis"
  />
</div>
`;

exports[`EventIcon when status: "available" and audience: "undefined" 1`] = `
<div
  className="pull-right"
>
  <Icon
    className="fa fa-calendar-check-o icon-green"
    message="Prenotabile"
  />
  <Icon
    className="hidden"
    message=""
  />
</div>
`;

exports[`EventIcon when status: "fully_booked" and audience: "gold" 1`] = `
<div
  className="pull-right"
>
  <Icon
    className="fa fa-calendar-times-o icon-red"
    message="Disponibilità Esaurita"
  />
  <Icon
    className="fa fa-trophy icon-yellow"
    message="Solo Corieristi Gold"
  />
</div>
`;

exports[`EventIcon when status: "fully_booked" and audience: "preferred" 1`] = `
<div
  className="pull-right"
>
  <Icon
    className="fa fa-calendar-times-o icon-red"
    message="Disponibilità Esaurita"
  />
  <Icon
    className="fa fa-star icon-yellow"
    message="Solo Corieristi Semper Fidelis"
  />
</div>
`;

exports[`EventIcon when status: "fully_booked" and audience: "undefined" 1`] = `
<div
  className="pull-right"
>
  <Icon
    className="fa fa-calendar-times-o icon-red"
    message="Disponibilità Esaurita"
  />
  <Icon
    className="hidden"
    message=""
  />
</div>
`;

exports[`EventIcon when status: "undefined" and audience: "gold" 1`] = `
<div
  className="pull-right"
>
  <Icon
    className="hidden"
    message=""
  />
  <Icon
    className="fa fa-trophy icon-yellow"
    message="Solo Corieristi Gold"
  />
</div>
`;

exports[`EventIcon when status: "undefined" and audience: "preferred" 1`] = `
<div
  className="pull-right"
>
  <Icon
    className="hidden"
    message=""
  />
  <Icon
    className="fa fa-star icon-yellow"
    message="Solo Corieristi Semper Fidelis"
  />
</div>
`;

exports[`EventIcon when status: "undefined" and audience: "undefined" 1`] = `
<div
  className="pull-right"
>
  <Icon
    className="hidden"
    message=""
  />
  <Icon
    className="hidden"
    message=""
  />
</div>
`;
